src/Form/DomainMenuAccessSettingsForm.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\domain_menu_access\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\system\Entity\Menu;
+
+class DomainMenuAccessSettingsForm extends ConfigFormBase {
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'domain_menu_access_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['domain_menu_access.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $menu = Menu::loadMultiple();
+    if (empty($menu)) {
+      $form['markup'] = [
+        '#markup' => $this->t('Your menu list is empty. Please, try add the menu and return here.'),
+      ];
+    } 
+    else {
+      $form['description'] = [
+        '#markup' => $this->t('Please, select menu for enable control by domain records.'),
+      ];
+      /** @var \Drupal\system\Entity\Menu $item */
+      foreach ($menu as $key => $item) {
+        $form[$key] = array(
+          '#type' => 'checkbox',
+          '#title' => $item->label(),
+          '#default_value' => $item->getThirdPartySetting('domain_menu_access', 'access_enabled'),
+          '#description' => $item->getDescription(),
+        );
+      }
+    }
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $menu = Menu::loadMultiple();
+    $values = $form_state->getValues();
+    /** @var \Drupal\system\Entity\Menu $item */
+    foreach ($menu as $key => $item) {
+      if ($values[$key] == 1) {
+        $item->setThirdPartySetting('domain_menu_access', 'access_enabled', TRUE);
+      }
+      else {
+        $item->setThirdPartySetting('domain_menu_access', 'access_enabled', FALSE);
+      }
+      $item->save();
```

Comment:
You need to check previous state, no reason to call save() if no changes (saves cache)